File: front/src/app/shared/models/client.model.ts
import { MODEL } from "..";
import { User } from "./user.model";
import { Address } from "./address.model";

export class Client extends User {
  constructor(
    public override id?: string,
    public override name?: string,
    public override cpf?: string,
    public override email?: string,
    public override password?: string,
    public override cellphone?: string,
    public override type?: string,
    public status?: string,
    public manager?: string,
    public salary?: string,
    public address?: MODEL.Address
  ) {
    super(id, name, cpf, email, password, cellphone, type);
  }
}

File: back/bantads/src/main/java/com/dac/bantads/model/ClientModel.java
package com.dac.bantads.model;

import jakarta.persistence.*;

@Entity
public class ClientModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String cpf;
    private String email;
    private String cellphone;
    private double salary;
    private String password;

    @Embedded
    private ClientModel address;


}

@Embeddable
public class ClientModel {
    private String cep;
    private String logradouro;
    private String bairro;
    private String localidade;
    private String uf;


}
